source/server/admin/admin.cc

```diff
@@ -307,39 +310,46 @@ class HandlerGasket : public Admin::Handler {
 
 } // namespace
 
-Admin::HandlerPtr AdminImpl::makeStaticTextHandler(absl::string_view response, Http::Code code) {
-  return std::make_unique<StaticTextHandler>(response, code);
+Admin::RequestPtr Admin::makeStaticTextRequest(absl::string_view response, Http::Code code) {
+  return std::make_unique<StaticTextRequest>(response, code);
+}
+
+Admin::RequestPtr Admin::makeStaticTextRequest(Buffer::Instance& response, Http::Code code) {
+  return std::make_unique<StaticTextRequest>(response, code);
 }
 
 Http::Code AdminImpl::runCallback(absl::string_view path_and_query,
                                   Http::ResponseHeaderMap& response_headers,
                                   Buffer::Instance& response, AdminStream& admin_stream) {
-  HandlerPtr handler = findHandler(path_and_query, admin_stream);
-  Http::Code code = handler->start(response_headers);
+  RequestPtr request = makeRequest(path_and_query, admin_stream);
+  Http::Code code = request->start(response_headers);
   bool more_data;
   do {
-    more_data = handler->nextChunk(response);
+    more_data = request->nextChunk(response);
   } while (more_data);
   Memory::Utils::tryShrinkHeap();
   return code;
 }
 
-Admin::HandlerPtr AdminImpl::findHandler(absl::string_view path_and_query,
+Admin::RequestPtr AdminImpl::makeRequest(absl::string_view path_and_query,
                                          AdminStream& admin_stream) {
   std::string::size_type query_index = path_and_query.find('?');
   if (query_index == std::string::npos) {
     query_index = path_and_query.size();
   }
 
+  // TODO(jmarantz): consider using an absl::btree_map or std::map rather than
```

Comment:
My guess is that until there are >50 routes to search, the linear search may actually be faster. I'm not sure this is even worth a TODO until it is shown that there's a problem here.